src/cli/repo/s3.go

```diff
@@ -232,12 +240,25 @@ func connectS3Cmd(cmd *cobra.Command, args []string) error {
 
 func S3Overrides() map[string]string {
 	return map[string]string{
-		config.AccountProviderTypeKey: account.ProviderM365.String(),
-		config.StorageProviderTypeKey: storage.ProviderS3.String(),
-		storage.Bucket:                bucket,
-		storage.Endpoint:              endpoint,
-		storage.Prefix:                prefix,
-		storage.DoNotUseTLS:           strconv.FormatBool(doNotUseTLS),
-		storage.DoNotVerifyTLS:        strconv.FormatBool(doNotVerifyTLS),
+		config.AccountProviderTypeKey:  account.ProviderM365.String(),
+		config.StorageProviderTypeKey:  storage.ProviderS3.String(),
+		credentials.AWSAccessKeyID:     utils.AWSAccessKeyFV,
+		credentials.AWSSecretAccessKey: utils.AWSSecretAccessKeyFV,
+		credentials.AWSSessionToken:    utils.AWSSessionTokenFV,
+		storage.Bucket:                 bucket,
+		storage.Endpoint:               endpoint,
+		storage.Prefix:                 prefix,
+		storage.DoNotUseTLS:            strconv.FormatBool(doNotUseTLS),
+		storage.DoNotVerifyTLS:         strconv.FormatBool(doNotVerifyTLS),
+	}
+}
+
+func S3UpdateFromEnvVar(s3Flag map[string]string) map[string]string {
```

Comment:
I'm a little concerned about how ownership is getting distributed across multiple packages in this change.  For example, env vars have moved [out of the config processing](https://github.com/alcionai/corso/pull/3590/files#diff-f37f5d414f19e717eb2ca53d62ece0c8fa1f4d85d88a1b3ce1bcc3218bc0775aL90) and here into the command itself.  That's not entirely bad, but I'm not sure what's gained here with the move, since no other packages seem to import `S3UpdateFromEnvVar` (which is correct, imo, no other packages should import it).  Seems to me like this could move back to `GetConfigRepoDetails`, and this func could go away, without any repercussions.

Reply:
Some S3 config like - Bucket, Endpoint are to be fetched from `envvar` only in case of `init` and `connect` command. And for all other cases it to to be fetched only from config file. So this override is done only in case of connect and init command.